### ISSUE TITLE
"supplementary" spelling typo in new Advanced option in Properties.

### ISSUE THREAD

**Issue by chip33** [valinet/ExplorerPatcher#346]
Love the new design and everything. 
Just spotted this typo

in "Supplimentary delay at logon". (there is a typo in Supplementary)

I checked the right spelling on Google to be sure :)

**Comment by valinet** [comment]
Yeah, it's been there for ages though, never noticed it. I mean, I never knew how to write it properly. So, yeah, I will push an update in a few minutes, see how these fancy new automatic updates work (hopefully they work) =)))

Thank you

**Comment by chip33** [comment]
you are very welcome :)

**Comment by valinet** [comment]
Thank you for the appreciative feedback. Just pushed the commit, a build should be released in a couple of minutes. Let's see if the infrastructure now holds up...

**Comment by valinet** [comment]
Yeah, for me, I chose to log off and then log back on, was notified that an update is available, went to "Properties" - "Updates" - "Install latest version", waited a bit for the download, UAC prompt, "Yes", Explorer restarts and "supplementary" is there. Nice :D

Ofc, the Windows toasts take too long to display, but whatever, it's important that it generally works. One day and idea will hit me and it's going to turn awesome. Like this setup/update program, I slept 10 hours a few days ago and when I woke up I had the idea, the infrastructure, everything how to set it up and just took a couple of days to get it off my head to the real world.

Yeah, I am off topic here. Thanks again for the tip.

**Comment by chip33** [comment]
hi update seemed to work well. 
I got a notification about being advised when download completed or something but didn't  see a notification after but it all happened so fast i couldnt be sure :)
It all happened so smoothly I wasn't able to really give a complete description but it definitely worked without an explorer refresh and the typo is fixed.

**Comment by valinet** [comment]
Yeah, the backend is pretty fast, Windows' toasts are too slow for it =))) After an update completes, there's no notification currently, it just works™. There is a popup when an update is available, notifying you, and another popup that shows after that one is dismissed or times out which tells you about the download progress. And that's it, then it just updates.

The thing it lacks is for a new toast to kill the old one. Like, if I popup a toast saying "new update", user hits update in the UI, I want to popup a toast saying "downloading", I do it in code, but it is queued by Windows and it will only show after the "new update" popup times out.

Balloons used to work the way I described above... Anyway, idk, as usual, I will think about it, at least it works fine now. I also want to write a bit in the wiki, all the links to it are empty pages now, I just made them so I do not have to issue updates whenever I actually write each of the articles there.

**Comment by plofhaan** [comment]
There's some more typos in the error dialog window, though... lol :)

![ExplorerPatcher error](https://user-images.githubusercontent.com/93003106/141696503-0b95fd46-43a3-41ad-9904-4a2fb0dc574b.png)

**Comment by valinet** [comment]
Besides "froma", what else is "broken" there?

**Comment by plofhaan** [comment]
Nothing! lol

**Comment by chip33** [comment]
also " if you have registered this as A shell extension" to be pedantic :)

**Comment by valinet** [comment]
> 
> 
> also " if you have registered this as A shell extension" :)

Is that really required? Anyway, can you check everything else as well, if we are doing this anyway...?

So 2 things by now:

* froma
* this as A shell extension

**Comment by chip33** [comment]
English is my native language, I will check it carefully..

I just found occured is wrong too it should be "occurred" which I didn't even notice. it may take me 10 minutes or so to do this properly and I will post it all here after.

**Comment by plofhaan** [comment]
Add Dark Theme support to the error dialog window maybe? ;]

**Comment by chip33** [comment]
An error has occurred while servicing this product.
This is most likely caused by one or more of the backup files from a previous update still being in use.
Unlocking the files will most likely fix this issue.

Troubleshooting steps:
* Close and reopen the "Properties" dialog if it is currently open.
* Kill and restart all "explorer.exe" processes.
* If you have registered this application as a shell extension, then restarting the computer will probably fix this.
* Lastly, reboot the computer and try  (the above steps) again.

-----------------------------------------------

Hi, i'm not saying this is perfect or that you couldn't tweak it later but as long as I have understood the bit about backup files from a previous update being blocked this is may be slightly clearer from a  native English speaker  point of view , others may disagree :)  it was good  anyway.

**Comment by chip33** [comment]
You could say 

**Unlocking the files should fix this issue.** 

to avoid repeating "most likely" it's up to you depending what you think.

**Comment by chip33** [comment]
also this 
**Lastly, reboot the computer and try (the above steps) again.**

could be 
**Lastly, reboot the computer and try again.**

depending on how you mean and how it goes wrong

**Comment by plofhaan** [comment]
> The thing it lacks is for a new toast to kill the old one.
"Step 9: Remove and manage notifications":
https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/send-local-toast-desktop-cpp-wrl#step-9-remove-and-manage-notifications

**Comment by valinet** [comment]
> 
> 
> > The thing it lacks is for a new toast to kill the old one.
> 
> "Step 9: Remove and manage notifications": https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/send-local-toast-desktop-cpp-wrl#step-9-remove-and-manage-notifications

How to integrate that with the existing code base...?

**Comment by valinet** [comment]
> 
> 
> An error has occurred while servicing this product. This is most likely caused by one or more of the backup files from a previous update still being in use. Unlocking the files will most likely fix this issue.
> 
> Troubleshooting steps:
> 
>     * Close and reopen the "Properties" dialog if it is currently open.
> 
>     * Kill and restart all "explorer.exe" processes.
> 
>     * If you have registered this application as a shell extension, then restarting the computer will probably fix this.
> 
>     * Lastly, reboot the computer and try  (the above steps) again.
> 
> 
> Hi, i'm not saying this is perfect or that you couldn't tweak it later but as long as I have understood the bit about backup files from a previous update being blocked this is may be slightly clearer from a native English speaker point of view , others may disagree :) it was good anyway.

Yeah, it is very good, I will add that it, thanks. I am tired, had i spent more time on it, it would have looked better from the get go. The thing is, the message initially read something like: "An error has occurred while servicing this product". Last minute I decided that's kind of unhelpful and prone to people opening issues not knowing what to try, when those tips will probably save at least 60-70% of potential escalated tickets. But yeah, I was in a hurry since I just pushed the initial broken update and was on the run to revise it as quickly as possible...

Anyway, thanks, I will do a push soon.

**Comment by chip33** [comment]
yes, i totally understand, it's really tricky doing clear tech descriptions. 
Really easy to miss things and then spot a better way the next day.

btw you have a double space after try in:
 **try  (the above steps) again.**

**Comment by valinet** [comment]
Probably fixed. I settled for this: "* Lastly, reboot the computer and try again.", as after a reboot replacing the file should not be a problem and the user can just retry and it will most likely just work. The errors are usually generated by even backups still being in use, which could happen if you apply 2 updates with the "Properties" dialog still open, or while using it as shell extension in which case it gets injected in most of the processes on the system.

**Comment by plofhaan** [comment]
> How to integrate that with the existing code base...?

No idea, but the top of the page mentions "classic non-packaged Win32 apps" so it seemed to me like this might be what you are looking for. I guess you would have to start with Step 1 on this same page I linked.

**Comment by valinet** [comment]
It's a bit more complicated than what that article suggests. Not speaking about the fact that I write in C and that's C++ with Microsoft's proprietary shenanigans on top (WRL), as yeah, there're ways around that too, but it's just not that "easy" as you said. It takes quite a few code. The current method is a bit hacky, as I skip most of the step they say it's "required", but I also have some limitations. Idk if this specifically can be circumvented, I just simply haven't looked it up. Thanks.

**Comment by valinet** [comment]
@plofhaan I retract my words, I already had it mostly implemented, just needed to add a few lines with my existing implementation, there is a "Hide" method for the individual toast. This was basically copy-paste from another project I worked almost 1 year ago on, so I forgot about all these details. Yeah, my sincere apologies, I am a bit tired with all the work and sometimes I become easily edgy like that. Yeah, so I implemented it, I can now retract a toast when I have to show another one, so it's way less confusing now. I will push an update soon. Again, sorry.
